.github/workflows/docs.yaml

```
@@ -22,6 +22,21 @@ on:
     - cron: '0 2 1 */2 SUN'  # every two months on Sunday at 2 am
 
 jobs:
+  pre_commit:
+    name: Run pre-commit hooks
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "1"
+      - name: set PY
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: pre-commit
+        uses: pre-commit/action@v3.0.0
   build_docs:
     name: Documentation
```

```suggestion
    name: Documentation
    needs: [pre_commit]
```

This will make, so the Documentation step is done if the pre-commit succeeds.